### DISCUSSION TITLE
Multipolygon 3d Self-intersection at or near point (29.226606777104532, 41.100948229847958, NaN)

### DISCUSSION THREAD

**Discussion by denizkocaboga** [NetTopologySuite/NetTopologySuite#558] [Q&A]
Hi,
i create a **multipolygon** geometry object from **wkt**  with using **wktreader**.
But created Multipolygon object is not valid. I don't understand why?

I looked with **NetTopologySuite.Operation.Valid.IsValidOp**. 
It's writing 
> Self-intersection at or near point (29.226606777104532, 41.100948229847958, NaN)

When i try to save this multipolygon to postgredb below error is occuring.
> side location conflict [ (29.226606032999754, 41.101011950020158, NaN) ]

My wkt is 3d. Polygons is intersect with with X,Y  but all Pollygons are different Z. So its intersection does not make sense.

**My Wkt :**
```
MULTIPOLYGON (((29.22655418 41.10106896 30.5, 29.2266131 41.10106936 30.5, 29.22661312 41.10106711 30.5, 29.22663514 41.10106725 30.5, 29.22663578 41.10101323 30.5, 29.22660602 41.10101303 30.5, 29.2266068 41.10094823 30.5, 29.2265556 41.10094789 30.5, 29.22655508 41.10099198 30.5, 29.22655925 41.10099203 30.5, 29.22655924 41.10099248 30.5, 29.22657353 41.10099258 30.5, 29.22657317 41.10102272 30.5, 29.22659341 41.10102285 30.5, 29.2265931 41.10104941 30.5, 29.22655441 41.10104914 30.5, 29.22655418 41.10106896 30.5)), ((29.22663645 41.1011085 27.45, 29.22663665 41.10109095 27.45, 29.22663487 41.10109092 27.45, 29.22663579 41.10101215 27.45, 29.22660603 41.10101195 27.45, 29.22660679 41.10094713 27.45, 29.22655561 41.10094678 27.45, 29.22655372 41.10110795 27.45, 29.22663645 41.1011085 27.45)))
```

My Code
```csharp
public static IGeometry ToGeometry(this string value)
{
            try
            {
                if (value.IsNullOrEmpty()) return null;

                WKTReader reader = new WKTReader();

                IGeometry result = reader.Read(value);

                bool isMultipolygonValid = result.IsValid; //it's return false

                return result;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
}
```

(edited by @airbreather: just tweaking formatting to help us read it)

**Comment by FObermaier** [comment]
As a rule of thumb, NTS (and JTS) operate on 2D geometries. Z ordinate values are carried along but are usually not taken into account for computation.

NTS (and JTS) adhere to the OGC SFS geometry model which means for a `MULTIPOLYGON` that its `POLYGON`s **may not** overlap and only touch at a finite number of points.
